pkg/agent/flowexporter/connections/conntrack_connections_perf_test.go

```diff
@@ -0,0 +1,180 @@
+// +build !race
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connections
+
+import (
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"math/big"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	connectionstest "antrea.io/antrea/pkg/agent/flowexporter/connections/testing"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+	interfacestoretest "antrea.io/antrea/pkg/agent/interfacestore/testing"
+	"antrea.io/antrea/pkg/agent/openflow"
+	proxytest "antrea.io/antrea/pkg/agent/proxy/testing"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+	k8sproxy "antrea.io/antrea/third_party/proxy"
+)
+
+const (
+	testNumOfConns        = 100000
+	testNumOfNewConns     = 10000
+	testNumOfDeletedConns = 10000
+)
+
+var (
+	connStore      *ConntrackConnectionStore
+	conns          []*flowexporter.Connection
+	mockConnDumper *connectionstest.MockConnTrackDumper
+)
+
+/*
+Sample output:
+    go test -test.v -run=BenchmarkPoll -test.benchmem -bench=. -memprofile memprofile.out -cpuprofile profile.out
+	goos: linux
+	goarch: amd64
+	pkg: antrea.io/antrea/pkg/agent/flowexporter/connections
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkPoll
+		conntrack_connections_perf_test.go:63:
+			Summary:
+			Number of initial connections: 100000
+			Number of new connections/poll: 10000
+			Number of deleted connections/poll: 10000
+	BenchmarkPoll-2   	       1	1406326459 ns/op	164436400 B/op	 3721614 allocs/op
+	PASS
+	ok  	antrea.io/antrea/pkg/agent/flowexporter/connections	1.581s
+*/
+
+func BenchmarkPoll(b *testing.B) {
+	disableLogToStderr()
+	setupConntrackConnStore(b)
+	for n := 0; n < b.N; n++ {
+		mockConnDumper.EXPECT().DumpFlows(uint16(openflow.CtZone)).Return(conns, testNumOfConns, nil)
+		connStore.Poll()
+		conns = generateUpdatedConns(conns)
+	}
+	b.Logf("\nSummary:\nNumber of initial connections: %d\nNumber of new connections/poll: %d\nNumber of deleted connections/poll: %d\n", testNumOfConns, testNumOfNewConns, testNumOfDeletedConns)
+}
+
+func setupConntrackConnStore(b *testing.B) {
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	mockIfaceStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false).AnyTimes()
```

Comment:
wouldn't it be more accurate to return Pod interface information? I think this can be postponed though, especially if you believe this part of `Poll` (`fillPodInfo`) is not CPU intensive

Reply:
Replaced with Pod interface instead of nil. It turns out that calling to `mockIfaceStore` has some CPU consumption. Maybe we should cache some ip-interface in future work.